src/Definition.php

```diff
@@ -6,69 +6,55 @@
  * @license   https://github.com/laminas/laminas-server/blob/master/LICENSE.md New BSD License
  */
 
+declare(strict_types=1);
+
 namespace Laminas\Server;
 
 use Countable;
 use Iterator;
+use Laminas\Server\Exception\InvalidArgumentException;
 
-/**
- * Server methods metadata
- */
 class Definition implements Countable, Iterator
 {
     /**
-     * @var array Array of \Laminas\Server\Method\Definition objects
+     * @var Method\Definition[]
      */
     protected $methods = [];
 
     /**
-     * @var bool Whether or not overwriting existing methods is allowed
+     * @var bool
      */
     protected $overwriteExistingMethods = false;
 
-    /**
-     * Constructor
-     *
-     * @param  null|array $methods
-     */
-    public function __construct($methods = null)
+    public function __construct(?array $methods = null)
     {
         if (is_array($methods)) {
             $this->setMethods($methods);
         }
     }
 
-    /**
-     * Set flag indicating whether or not overwriting existing methods is allowed
-     *
-     * @param mixed $flag
-     * @return \Laminas\Server\Definition
-     */
-    public function setOverwriteExistingMethods($flag)
+    public function setOverwriteExistingMethods(bool $flag): self
     {
-        $this->overwriteExistingMethods = (bool) $flag;
+        $this->overwriteExistingMethods = $flag;
         return $this;
     }
 
     /**
      * Add method to definition
      *
-     * @param  array|\Laminas\Server\Method\Definition $method
+     * @param  array|Method\Definition $method
      * @param  null|string $name
-     * @return \Laminas\Server\Definition
-     * @throws \Laminas\Server\Exception\InvalidArgumentException if duplicate or invalid method provided
+     * @return $this
+     * @throws InvalidArgumentException if duplicate or invalid method provided
      */
-    public function addMethod($method, $name = null)
+    public function addMethod($method, ?string $name = null): self
     {
         if (is_array($method)) {
             $method = new Method\Definition($method);
         } elseif (! $method instanceof Method\Definition) {
             throw new Exception\InvalidArgumentException('Invalid method provided');
         }
 
-        if (is_numeric($name)) {
-            $name = null;
-        }
```

Comment:
string can be numeric, eg : 

```php
➜  ~ php -a
Interactive shell

php > var_dump(is_string('1'));
php shell code:1:
bool(true)

php > var_dump(is_numeric('1'));
php shell code:1:
bool(true)
```

Reply:
ACK. Add check for numeric string and reset if so (see 58c9e97).